src/requests/actions.jl

```diff
@@ -43,8 +55,12 @@ function textDocument_codeAction_request(params::CodeActionParams, server::Langu
         for (_, sa) in LSActions
             if sa.when(x, params)
                 action = CodeAction(
-                    sa.desc, sa.kind, missing, missing, missing,
-                    Command(sa.desc, sa.id, arguments),
+                    sa.desc, # title
+                    sa.kind, # kind
+                    missing, # diagnostics
+                    client_preferred_support(server) ? sa.preferred : missing, # isPreferred
```

Comment:
Perhaps it is too "strict" to only send this if the client claims to support it -- servers and clients should ignore unknown fields anyway IIUC.